test/DocumentFormat.OpenXml.Packaging.Tests/OpenXmlPackageTests.cs

```diff
@@ -301,5 +304,31 @@ public void TestOpenModel3DWrittenByPowerPoint_DashMime()
 
             Assert.Equal("model/gltf-binary", model3DReferenceRelationshipPart.ContentType);
         }
+
+        // Test issue #1281 for regressions.
+        // When opening a workbook (SpreadsheetDocument.Open) with a missing calcChain part and default OpenSettings
+        // specified, we should throw an exception.
+        [Fact]
+        public void OpenWithMissingCalcChainPart()
+        {
+            Stream stmSpd = GetStream(TestFiles.MissingCalcChainPart, false);
+            Action act = () => SpreadsheetDocument.Open(stmSpd, false);
+            InvalidOperationException exception = Assert.Throws<InvalidOperationException>(act);
+
+            Assert.Equal("Specified part does not exist in the package.", exception.Message);
+
+            try
+            {
+                SpreadsheetDocument spd = SpreadsheetDocument.Open(stmSpd, false, new OpenSettings() { IgnoreExceptionOnCalcChainPartMissing = true });
+
+                Assert.NotNull(spd);
+
+                spd.Close();
+            }
+            catch (InvalidOperationException ex)
```

Comment:
Why not do this in an Assert.Throws?

Reply:
I actually want to assert that it doesn't throw. but maybe I don't need to catch this, just let it throw?

Reply:
prosed new code, no try/catch